scripts/pylib/twister/twisterlib/testplan.py

```diff
@@ -418,6 +418,18 @@ def add_configurations(self):
         platform_config = self.test_config.get('platforms', {})
         for folder in board_dirs:
             for file in glob.glob(os.path.join(folder, "*.yaml")):
+                # If the user set a platform filter, we can, if no other option would increase
+                # the allowed platform pool, save on time by not loading YAMLs of any boards
+                # that do not start with the required names.
+                if self.options.platform and \
+                    not self.options.all and \
+                    not self.options.integration and \
+                    not any([
+                        os.path.basename(file).startswith(
+                            re.split('[/@]', p)[0]
+                        ) for p in self.options.platform
+                    ]):
+                    continue
```

Comment:
how does this work with hwmv1 boards?

Reply:
[Old porting guide](https://docs.zephyrproject.org/3.6.0/hardware/porting/board_porting.html#board-porting-guide) is compliant, as it tells users to name the `.yaml` the same way as their board name. However, the [Twister Runner docs](https://docs.zephyrproject.org/3.6.0/develop/test/twister.html#board-configuration) do not repeat that stipulation, e.g. only mentioning the Twister call for a revision (`board@revision`, which would work), rather than specifying relevant filenames.

As the docs seem insufficient, I can go back to before HWMv1 and check whether the assumption used here holds in practice.

Reply:
I have checked it on commit 418b0a1e65, parent of the HWMv2 commit (8dc3f856229ce083c956aa301c31a23e65bd8cd8).

All but three boards follow the pattern of `<boardname>.yaml`. The three outliers follow the pattern `<boardname>@<revision>.yaml` (
`stm32f3_disco@E` in `stm32f3_disco_E.yaml`, `s32z270dc2_rtu1_r52@D` in `s32z270dc2_rtu1_r52_D.yaml`, `s32z270dc2_rtu0_r52@D` in `s32z270dc2_rtu0_r52_D.yaml` )

I think this is sufficient to conclude that the assumption holds and this will work for HWMv1.